### PR TITLE
[Kernel] Clean up package organization of `kernel-default` module.

### PR DESCRIPTION
**Pull request by vkorukanti**
## Description
Rename the module `kernel-default` to `kernel-defaults`.
Rename package `io.delta.kernel` to `io.delta.kernel.defaults` in `kernel-defaults` module.
Move non-public classes under the `io.delta.kernel.defaults.internal` package.

Note: Reason for not using the `default` is because it is a keyword in Java and can't be part of the package namespace.

## How was this patch tested?
Updated the `kernel/examples` and ran the `python3 kernel/examples/run-kernel-examples.py --use-local` successfully.

### CODE REVIEW

File: kernel/kernel-default/src/main/java/io/delta/kernel/defaults/client/DefaultExpressionHandler.java [hunk 1]
```diff
@@ -14,24 +14,38 @@
...
 import io.delta.kernel.data.Row;
-import io.delta.kernel.data.vector.DefaultBooleanVector;
-import io.delta.kernel.data.vector.DefaultConstantVector;
+import io.delta.kernel.defaults.internal.data.vector.DefaultBooleanVector;
+import io.delta.kernel.defaults.internal.data.vector.DefaultConstantVector;
 import io.delta.kernel.expressions.Expression;
 import io.delta.kernel.expressions.ExpressionEvaluator;
 import io.delta.kernel.expressions.Literal;
-import io.delta.kernel.types.*;
+import io.delta.kernel.types.BinaryType;
```
Comment (vkorukanti) [inline kernel/kernel-default/src/main/java/io/delta/kernel/defaults/client/DefaultExpressionHandler.java:32]: We don't have a checkstyle rule to collapse these together.

**Comment by jaceklaskowski** [comment]
Please use `io.delta.kernel.default` (singular not plural) 🙏 That would make more sense to me given the package name is `kernel-default`. Just like `org.apache.spark.sql` not `org.apache.spark.sqls` 😉

**Comment by vkorukanti** [comment]
> Please use `io.delta.kernel.default` (singular not plural) 🙏 That would make more sense to me given the package name is `kernel-default`. Just like `org.apache.spark.sql` not `org.apache.spark.sqls` 😉

Agree it is big odd, unfortunately `default` is a keyword in Java and can't be used within the package namespace. Updated the PR to rename the module to `kernel-defaults` so that they are in sync. Also this module contains default implementations of multiple clients such as `DefaultExpressionHandler`, `DefaultParquetHandler` etc. Let me know if there are any issues or better naming alternatives.

File: kernel/examples/table-reader/src/main/java/io/delta/kernel/examples/BaseTableReader.java [hunk 2]
```diff
@@ -31,17 +34,15 @@
...
 import io.delta.kernel.data.ColumnVector;
 import io.delta.kernel.data.ColumnarBatch;
 import io.delta.kernel.data.DataReadResult;
-import io.delta.kernel.data.vector.VectorUtils;
 import io.delta.kernel.types.StructField;
 import io.delta.kernel.types.StructType;
 
-import static java.lang.String.format;
-import static java.util.Objects.requireNonNull;
+import io.delta.kernel.defaults.internal.data.vector.VectorUtils;
```
Comment (allisonport-db) [inline kernel/examples/table-reader/src/main/java/io/delta/kernel/examples/BaseTableReader.java:44]: nit: internal second?

File: kernel/kernel-defaults/src/test/java/io/delta/kernel/client/TestDefaultExpressionHandler.java
Comment (allisonport-db) [inline kernel/kernel-defaults/src/test/java/io/delta/kernel/client/TestDefaultExpressionHandler.java:1]: Should tests also be moved to io.delta.kernel.defaults?

File: kernel/kernel-defaults/src/test/java/io/delta/kernel/defaults/client/TestDefaultJsonHandler.java [hunk 3]
```diff
@@ -40,7 +42,10 @@
 import io.delta.kernel.types.StructType;
 import io.delta.kernel.utils.CloseableIterator;
 import io.delta.kernel.utils.Utils;
-import static io.delta.kernel.utils.DefaultKernelTestUtils.getTestResourceFilePath;
+
+import io.delta.kernel.defaults.utils.DefaultKernelTestUtils;
```
Comment (allisonport-db) [inline kernel/kernel-defaults/src/test/java/io/delta/kernel/defaults/client/TestDefaultJsonHandler.java:46]: nit: no new line? we can't actually enforce this I think

File: kernel/kernel-defaults/src/test/java/io/delta/kernel/defaults/integration/BaseIntegration.java [hunk 4]
```diff
@@ -32,22 +26,18 @@
...
-import io.delta.kernel.types.DecimalType;
-import io.delta.kernel.types.MapType;
-import io.delta.kernel.types.StructField;
-import io.delta.kernel.types.StructType;
-import io.delta.kernel.types.TimestampType;
+import io.delta.kernel.types.*;
 import io.delta.kernel.utils.CloseableIterator;
-import io.delta.kernel.utils.DefaultKernelTestUtils;
+
+import io.delta.kernel.defaults.client.DefaultTableClient;
```
Comment (allisonport-db) [inline kernel/kernel-defaults/src/test/java/io/delta/kernel/defaults/integration/BaseIntegration.java:37]: same thing? I think you can update this in the settings

File: kernel/kernel-defaults/src/test/java/io/delta/kernel/defaults/integration/TestDeltaTableReads.java [hunk 5]
```diff
@@ -26,14 +26,16 @@
...
-import static io.delta.kernel.DefaultKernelUtils.daysSinceEpoch;
-import static io.delta.kernel.integration.DataBuilderUtils.row;
-import static io.delta.kernel.utils.DefaultKernelTestUtils.getTestResourceFilePath;
+
+import io.delta.kernel.defaults.client.DefaultTableClient;
+import io.delta.kernel.defaults.integration.DataBuilderUtils.TestColumnBatchBuilder;
+import static io.delta.kernel.defaults.integration.DataBuilderUtils.row;
+import static io.delta.kernel.defaults.utils.DefaultKernelTestUtils.getTestResourceFilePath;
+
+import static io.delta.kernel.defaults.internal.DefaultKernelUtils.daysSinceEpoch;
```
Comment (allisonport-db) [inline kernel/kernel-defaults/src/test/java/io/delta/kernel/defaults/integration/TestDeltaTableReads.java:38]: nit same thing

File: kernel/kernel-defaults/src/test/scala/io/delta/kernel/DeletionVectorSuite.scala
Comment (allisonport-db) [inline kernel/kernel-defaults/src/test/scala/io/delta/kernel/DeletionVectorSuite.scala:1]: move this to /defaults/ as well?

File: kernel/kernel-defaults/src/test/java/io/delta/kernel/defaults/client/TestDefaultJsonHandler.java (hunk 3 above)
Comment (vkorukanti) [reply kernel/kernel-defaults/src/test/java/io/delta/kernel/defaults/client/TestDefaultJsonHandler.java:46]: I modified the import order in Intellij to remove the blank line.

**Review by allisonport-db — APPROVED** [review]
LGTM

**Pull request merged**